### PR TITLE
Use tRPC v10 in the guestbook-blog #10

### LINKED ISSUE #10: Use tRPC v10 in the guestbook-blog
Refer: https://github.com/nexxeln/guestbook-blog/issues/1

### PR DESCRIPTION
**Pull request by AyanavaKarmakar**
Closes #10

### CODE REVIEW

**Comment by nexxeln** [comment]
Hey! Thanks for doing this. There seems to be problem in egoist's gh pinned repos api, thats why its failing the deployment. I'll try to fix this and then make this work on a preview deployment to review. This might take some time though. At first glance looks pretty good.

**Comment by AyanavaKarmakar** [comment]
Sounds good!

**Comment by nexxeln** [comment]
Fixed the errors, had to create my own api.

Please update your branch with the latest commit so we can get a preview deployment working

**Comment by AyanavaKarmakar** [comment]
@nexxeln re-authorize deployment whenever you're free.

File: content/ct3a-guestbook.mdx [hunk 1]
```diff
@@ -200,92 +198,128 @@ Let's push this modified schema to our Railway Postgres database.
...
-    // Any queries or mutations after this middleware will
-    // raise an error unless there is a current session
-    if (!ctx.session) {
-      throw new TRPCError({ code: "UNAUTHORIZED" });
-    }
-    return next();
-  })
-  .mutation("postMessage", {
-    // ...
+If you take a look at the `src/server/common/get-server-auth-session.ts` file, we're using [`unstable_getServerSession`] from Next Auth that gives us access to the session on the server. We're passing that into our tRPC context. We can use this session to make our mutation protected using a `protectedProcedure`.
```
Comment (juliusmarminge) [inline content/ct3a-guestbook.mdx]: This is a bit un-cohesive. First mentioning the `get-server-auth-session.ts` and then showing the middleware from `trpc.ts`.

File: content/ct3a-guestbook.mdx [hunk 2]
```diff
@@ -200,92 +198,128 @@ Let's push this modified schema to our Railway Postgres database.
...
+    },
+  });
+});
+
+export const protectedProcedure = t.procedure.use(isAuthed);
 ```
 
-Next, let's write a query to get all messages in the guestbook.
-This doesn't need to be protected so we can put it above the middleware.
+Now, replace `publicProcedure` with `protectedProcedure` to make our mutation protected.
```
Comment (juliusmarminge) [inline content/ct3a-guestbook.mdx]:
```suggestion
Now, replace `publicProcedure` with `protectedProcedure` to make our mutation protected from unauthenticated users.
```

File: content/ct3a-guestbook.mdx [hunk 3]
```diff
@@ -200,92 +198,128 @@ Let's push this modified schema to our Railway Postgres database.
...
-    },
-  })
-  .middleware(async ({ ctx, next }) => {
-    // ...
+    }),
+})
+```
+
+Next, let's write a query to get all messages in the guestbook.
+This doesn't need to be protected so we can put it above the middleware.
```
Comment (juliusmarminge) [inline content/ct3a-guestbook.mdx]: not relevant anymore
```suggestion
```

File: content/ct3a-guestbook.mdx [hunk 4]
```diff
@@ -200,92 +198,128 @@ Let's push this modified schema to our Railway Postgres database.
...
       }
-    },
-  })
-  .middleware(async ({ ctx, next }) => {
-    // ...
+    }),
+})
+```
+
+Next, let's write a query to get all messages in the guestbook.
```
Comment (juliusmarminge) [inline content/ct3a-guestbook.mdx]:
```suggestion
Next, let's write a query to get all messages in the guestbook. We wan't all guests to see the messages so we'll use the `publicProcedure` for this.
```

File: content/ct3a-guestbook.mdx [hunk 5]
```diff
@@ -200,92 +198,128 @@ Let's push this modified schema to our Railway Postgres database.
...
 
 Now merge this router in the main `appRouter`.
 
-```ts:src/server/router/index.ts
-import superjson from "superjson";
-import { createRouter } from "./context";
-import { guestbookRouter } from "./guestbook";
+```ts:src/server/trpc/router/_app.ts
+import { router } from "../trpc";
+import { authRouter } from "./auth.ts"
```
Comment (juliusmarminge) [inline content/ct3a-guestbook.mdx]: I'd remove this since it's not used by the guestbook
```suggestion
```

File: content/ct3a-guestbook.mdx [hunk 6]
```diff
@@ -200,92 +198,128 @@ Let's push this modified schema to our Railway Postgres database.
...
+import { guestbookRouter } from "./guestbook.ts"
 
-export const appRouter = createRouter()
-  .transformer(superjson)
-  .merge("guestbook.", guestbookRouter);
+export const appRouter = router({
+  auth: authRouter,
```
Comment (juliusmarminge) [inline content/ct3a-guestbook.mdx]:
```suggestion
```

File: content/ct3a-guestbook.mdx [hunk 7]
```diff
@@ -496,17 +538,17 @@ For this we can use optimistic UI updates! `react-query` makes this trivial to d
 
 ```ts:src/pages/index.tsx
 const ctx = trpc.useContext();
```
Comment (juliusmarminge) [inline content/ct3a-guestbook.mdx]: utils is the conventional variable name for the trpc 10 context

**Review by juliusmarminge — COMMENTED** [review]
CleanShot 2022-10-27 at 20 04 33

File: content/ct3a-guestbook.mdx (hunk 7 above)
Comment (nexxeln) [reply content/ct3a-guestbook.mdx]: oh i didn't know this, why tho? doesn't ctx refer to context?

File: content/ct3a-guestbook.mdx [hunk 8]
```diff
@@ -5,8 +5,6 @@ date: "2022-08-11"
 draft: false
 ---
 
-> Update: This blog post was written before [tRPC v10](https://trpc.io/docs/v10/) was released. You should use tRPC v10 now.
```
Comment (vlmlee) [inline content/ct3a-guestbook.mdx:8]: Should probably add a line that this tutorial has been updated, ideally by the date above with `Updated on Oct 27, 2022`, etc. I'm not familiar with this repo so I don't know if `updated` is part of the blog post model.

File: content/ct3a-guestbook.mdx (hunk 7 above)
Comment (juliusmarminge) [reply content/ct3a-guestbook.mdx]: tbh im not sure xD

**Review by vlmlee — CHANGES_REQUESTED** [review]

File: content/ct3a-guestbook.mdx (hunk 8 above)
Comment (nexxeln) [reply content/ct3a-guestbook.mdx:8]: There's no updated field, a blockquote on saying its updated is a good addition imo.

**Review by vlmlee — APPROVED** [review]
Changes look good. Code examples are clear and allows people to see the entire plumbing.

**Review by juliusmarminge — APPROVED** [review]
LGTM

**Comment by nexxeln** [comment]
Thanks so much! @AyanavaKarmakar @juliusmarminge @vlmlee

**Pull request merged**